app/models/submission.rb

```diff
@@ -0,0 +1,54 @@
+require 'securerandom'
+
+class Submission < ActiveRecord::Base
+  belongs_to :project
+  has_many :jobs, dependent: :destroy
+
+  store :job_attrs
+
+  attr_accessor :start_frame, :end_frame, :project_dir
+
+  def submit
+    parse_frames
+
+    job = new_job
+    success = job.submit(templated_content)
+
+    success
+  end
+
+  def never_submitted_status
+    'not submitted'
+  end
+
+  private
+
+  def submission_template
+    'jobs/video_jobs/maya_submit.sh.erb'
+  end
+
+  def new_job
+    job = Job.create(
+      job_id: 'Never-Submitted-' + SecureRandom.urlsafe_base64(5),
+      status: never_submitted_status,
+      submission_id: attributes['id'],
+      cluster: attributes['cluster']
+    )
+    job
```

Comment:
Setting a local job variable here is unnecessary: Below is preferable

```ruby
def new_job
  Job.create(
    ...
  )
end
```

Reply:
Done, but also using the Job.new api instead.

Reply:
I thought I could go to new api, but turns out not.

I persist it first to get a database ID. That's part of the jobs working directory `$HOME/ondemand/data/.../$DB_ID/`.  This is the directory where I'm keeping the output logs and shell script that was submitted, etc. 

I could generate some other ID, then store maybe the job_work_dir?